plugins/outputs/cloud_pubsub/cloud_pubsub.go

```diff
@@ -56,17 +58,16 @@ func (ps *PubSub) SetSerializer(serializer serializers.Serializer) {
 }
 
 func (ps *PubSub) Connect() error {
-	if ps.Topic == "" {
-		return fmt.Errorf(`"topic" is required`)
-	}
-
-	if ps.Project == "" {
-		return fmt.Errorf(`"project" is required`)
+	var err error
+	ps.encoder, err = internal.NewContentEncoder(ps.ContentEncoding)
+	if err != nil {
+		return err
 	}
```

Comment:
Same as for the input. We only need the `encoder` if not `identity`...

Reply:
Moved to a separate if statement underneath the switch statement, as to too much nesting of 'if' statements.